ecommerce/views.py

```diff
@@ -201,6 +232,75 @@ def patch(self, request, *args, **kwargs):
         )
 
 
+class CouponView(APIView):
+    """
+    Admin view for creating coupon(s)
+    """
+
+    permission_classes = (IsAdminUser,)
+    authentication_classes = (SessionAuthentication,)
+
+    def post(self, request, *args, **kwargs):
+        """ Create coupon(s) and related objects """
+        # Determine what kind of coupon this is.
+        if request.data.get("coupon_type") == CouponPaymentVersion.SINGLE_USE:
+            coupon_serializer = SingleUseCouponSerializer(data=request.data)
+        else:
+            coupon_serializer = PromoCouponSerializer(data=request.data)
+        if coupon_serializer.is_valid():
+            coupon_data = coupon_serializer.validated_data
+            with transaction.atomic():
```

Comment:
This kind of logic really should be in the serializer itself since that aligns with typical DRF usage

Reply:
👍